dashboard/app/controllers/authored_hint_view_requests_controller.rb

```diff
@@ -9,7 +9,31 @@ def create
       return head :bad_request
     end
 
-    hints = params.permit(hints: [:scriptId, :levelId, :hintId]).require(:hints)
+    fields = [
+      :scriptId,
+      :levelId,
+      :hintId,
+      :hintClass,
+      :hintType,
+      :prevTime,
+      :prevAttempt,
+      :prevTestResult,
+      :prevActivityId,
+      :prevLevelSourceId,
+      :nextTime,
+      :nextAttempt,
+      :nextTestResult,
+      :nextActivityId,
+      :nextLevelSourceId,
+      :finalTime,
+      :finalAttempt,
+      :finalTestResult,
+      :finalActivityId,
+      :finalLevelSourceId,
+    ]
+
+    hints = params.permit(hints: fields).require(:hints)
```

Comment:
Note that my migration in #13232 will cause the activity references to no longer be necessary. For clarity, I'll add dropping those columns to the migration PR.